### PR TITLE
[ticket/17321] Add versioning to push worker to force updates

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB-17321

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB-17321

### CODE REVIEW

File: phpBB/phpbb/ucp/controller/webpush.php [hunk 1]
```diff
@@ -138,6 +138,7 @@ public function worker(): Response
 		// @todo: only work for logged in users, no anonymous & bot
 		$content = $this->template->render('push_worker.js.twig', [
 			'U_WEBPUSH_GET_NOTIFICATION'	=> $this->controller_helper->route('phpbb_ucp_push_get_notification_controller'),
+			'ASSETS_VERSION'				=> $this->config['assets_version'],
```
Comment (iMattPro) [inline phpBB/phpbb/ucp/controller/webpush.php:141]: `config` is not defined, needs to be added to the class

**Comment by iMattPro** [comment]
This works and I think is better than the other way!

File: phpBB/styles/all/js/push_worker.js.twig [hunk 2]
```diff
@@ -1,3 +1,15 @@
+/**
+ * Event listener for install event
+ */
+self.addEventListener('install', () => {
+	// Call to ensure service worker is correctly updated
+	self.skipWaiting();
+});
+
+self.addEventListener('activate', event => {
+	event.waitUntil(clients.claim());
```
Comment (rxu) [inline phpBB/styles/all/js/push_worker.js.twig:10]:
```suggestion
	event.waitUntil(self.clients.claim());
```

**Comment by rxu** [comment]
Tested and it worked like that:
- changed worker;
- purged the cache;
- sent webpush from account 1 to account 2 - no changes;
- sent webpush from account2 to account1 - no changes;
- sent webpush from account 1 to account2 - got changed webpush;
- sent webpush from account2 to account1 - got changed webpush.

So it works.

**Pull request merged**